trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java

```diff
@@ -1138,4 +1170,94 @@ private static Response validateParameters(String outputId, QueryParameters quer
+     * @return a list of data provider descriptors
+     */
+    @POST
+    @Path("/{outputId}")
+    @Tag(name = OCG)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Get the list of outputs for this configuration", responses = {
+            @ApiResponse(responseCode = "200", description = "Returns a list of output provider descriptors", content = @Content(array = @ArraySchema(schema = @Schema(implementation = DataProvider.class)))),
+            @ApiResponse(responseCode = "400", description = INVALID_PARAMETERS, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = PROVIDER_NOT_FOUND, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = NO_SUCH_CONFIGURATION, content = @Content(schema = @Schema(implementation = String.class))),
+    })
+    public Response createDataProvider(
+                @Parameter(description = EXP_UUID) @PathParam("expUUID") UUID expUUID,
+                @Parameter(description = OUTPUT_ID) @PathParam("outputId") String outputId,
+                @RequestBody(description = CFG_CREATE_DESC + " " + CFG_KEYS_DESC, content = {
+                        @Content(schema = @Schema(implementation = org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model.OutputConfigurationQueryParameters.class))
+                }, required = true) OutputConfigurationQueryParameters queryParameters) {
+        return Response.status(Status.NOT_IMPLEMENTED).entity("POST a configuration to a given output to create derived output").build(); //$NON-NLS-1$
+    }
+
+    /**
+     * DELETE a derived data provider from created by a given data provider
+     *
+     * @param expUUID
+     *            desired experiment UUID
+     * @param outputId
+     *            Output ID for the data provider to query
+     * @param derivedOutputId
+     *            Output ID for the data provider to delete
+     * @return status and the deleted configuration instance, if successful
+     */
+    @DELETE
+    @Path("/{outputId}/{derivedOutputId}")
+    @Tag(name = OCG)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Delete a configuration instance of a given configuration source type", responses = {
+            @ApiResponse(responseCode = "200", description = "The derived data provider (and it's configuration) was successfully deleted", content = @Content(schema = @Schema(implementation = org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model.Configuration.class))),
+            @ApiResponse(responseCode = "404", description = PROVIDER_NOT_FOUND, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = NO_SUCH_CONFIGURATION, content = @Content(schema = @Schema(implementation = String.class))),
```

Comment:
I assume PROVIDER_NOT_FOUND is for the outputId and NO_SUCH_CONFIGURATION is for the derivedOutputId?
Should we have a different message for that one?

Reply:
yes, if derived output doesn't exists then we would return NO_SUCH_CONFIGURATION.